scripts/validate-signature.ts

```diff
@@ -0,0 +1,77 @@
+import { readFileSync } from 'fs';
+import { ClientRequest } from 'http';
+import { parseRequest, ParseOptions, verifySignature } from 'http-signature';
+import { parseCertificate } from 'sshpk';
+
+const getRequiredQsealHeaders = (): string[] => [
+  '(request-target)',
+  'date',
+  'psu-ip-address',
+  'psu-ip-port',
+  'psu-http-method',
+  'psu-date',
+  'psu-user-agent',
+  'psu-referer',
+  'psu-accept',
+  'psu-accept-charset',
+  'psu-accept-encoding',
+  'psu-accept-language',
+];
+
+const CERTIFICATE_FORMAT = 'pem';
+
+export type Body = {
+  method: string;
+  url: string;
+  headers: Record<string, string | number>;
+};
+
+const validateSignature = async (body: Body, certificate: string) => {
+  const parsedSignature = parseRequest(
+    body as unknown as ClientRequest,
+    {
+      authorizationHeaderName: 'signature',
+      headers: getRequiredQsealHeaders(),
+      strict: true,
+    } as unknown as ParseOptions,
+  );
+  const signatureKeyId = parsedSignature.params.keyId;
+
+  const parsedCertificate = parseCertificate(certificate, CERTIFICATE_FORMAT);
+  const { subjectKey: publicKey } = parsedCertificate;
+
+  const result = verifySignature(parsedSignature, publicKey.toString(CERTIFICATE_FORMAT));
+  console.log('Signature verified result for signatureKeyId:', signatureKeyId);
+  console.log(result);
+};
+
+const req: Body = {
+  method: 'GET',
+  url: '/users/profiles/e719f43e-a9bb-4dd8-bf61-68c0b251250e',
+  headers: {
+    public_load_balancer: 'false',
+    client_cert_present: 'true',
+    client_cert_chain_verified: 'true',
+    'psu-accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
+    'psu-accept-charset': 'utf-8',
+    'psu-accept-encoding': 'gzip, deflate, br',
+    'psu-accept-language': 'en-US,en;q=0.9',
+    'psu-date': '2019-01-01T00:00:00Z',
+    'psu-http-method': 'GET',
+    'psu-ip-address': '127.0.0.1',
+    'psu-ip-port': 443,
+    'psu-referer': 'https://localhost:443/',
+    'psu-user-agent':
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36',
+    authorization:
```

Comment:
Why is there a hardcoded token here? Since this repo is public we can consider it as leaked